### ISSUE TITLE
API Gateway {proxy+} event inconsistent with AWS 

### ISSUE THREAD

**Issue by stevenj** [localstack/localstack#2210]
# Type of request: This is a ...

[X] bug report
[ ] feature request

# Detailed description
I deployed a simple lambda app using {proxy+} routing for API Gateway.

The event delivered to the lambda function diverges from the one the AWS Gateway delivers in a number of areas, which breaks my lambda function.

## Expected behaviour

The Events generated by APIGateway should reasonably reflect the event produced by AWS Api Gateway.

### Serverless Template
```yaml
service: Test
plugins:
  - serverless-python-requirements
  - serverless-localstack
custom:
  pythonRequirements:
    dockerizePip: non-linux
  localstack:
    debug: true
    stages:
      - local
    host: http://localhost
    autostart: true
    endpoints:
      S3: http://localhost:4572
      DynamoDB: http://localhost:4569
      CloudFormation: http://localhost:4581
      Elasticsearch: http://localhost:4571
      ES: http://localhost:4578
      SNS: http://localhost:4575
      SQS: http://localhost:4576
      Lambda: http://localhost:4574
      Kinesis: http://localhost:4568
    lambda:
      mountCode: False
    docker:
      sudo: False

provider:
  name: aws
  runtime: python3.8

  stage: dev
  region: us-east-1

package:
  individually: true
  exclude:
    - "*/**"
    - "*"

functions:
  API_Dispatch:
    handler: mothership.API_Dispatch.lambda_handler
    module: backend/serverless
    package:
      include:
        - backend/serverless/**
    events:
      - http: GET /
      - http: 'GET {proxy+}'
      - http: POST /
      - http: 'POST {proxy+}'
```

### Significant Differences in the events generated

AWS:         `"resource": "/{proxy+}",`
localstack: `"resource": "/restapis/q9g0ziob0l/local/_user_request_/v1/bounce",`

localstack should set `"resource"` to `"{/proxy+}"`

AWS: `"multiValueHeaders": { ... }`
localstack: Does not produce `multiValueHeaders` in the event

AWS: `""queryStringParameters": null,`
localstack: `"queryStringParameters": {},`

`queryStringParameters` should be `null` if no parameters sent.

AWS: `"multiValueQueryStringParameters": null,`
localstack: Does not produce `multiValueQueryStringParameters` in the event

AWS: `"pathParameters": { "proxy": "v1/bounce" },`
localstack: `"pathParameters": { "proxy+": "v1/bounce" },`

Key should be `proxy` not `proxy+`

AWS: `"stageVariables": null,`
localstack: `"stageVariables": {},`

`stageVariables` should be null if no variables set.

AWS: `"requestContext": { "path": "/dev/v1/bounce", }`
localstack: `"requestContext": {"path": "/v1/bounce", }`

First part of path in `requestContext` should be the stage.

There is a lot missing from `requestContext` but notable entries are:
```json
"requestContext": {
    "httpMethod": "GET",
    "requestTime": "25/Mar/2020:04:53:40 +0000",
    "protocol": "HTTP/1.1",
    "requestTimeEpoch": 1585112020858,
```
All of these should be able to be generated with information the API Gateway should have available.  And would make the events more consistent.

# Steps to reproduce

1. Create an lambda function that is called by API Gateway using {proxy+}
2. Deploy Locally and to AWS
3. Compare the events received by the lambda function.

## Command used to start LocalStack

```serverless deploy --stage local```

**Comment by derKrischan** [comment]
I can confirm the report above. When trying to build an integration test for a Java based lambda with localstack the headers are never "forwarded" to the lambda because the Quarkus handler expects multiValueHeaders instead of "headers" as sent by localstack API Gateway and "multiValueQueryStringParameters" instead of "queryStringParameters".

**Comment by derKrischan** [comment]
Is this topic more related to moto? My python skills are pretty poor but perhaps we should open an issue there.

**Comment by whummer** [comment]
Thanks for reporting @stevenj . The majority of issues should be fixed via #2526 and #2557 . Can you please give it another try with the `latest` version of the Docker image? Please let us know if any of these issues still persist. Thanks
